src/controller/testController.ts

```diff
@@ -265,23 +265,57 @@ function enqueueTestMethods(testItems: TestItem[], run: TestRun): void {
  * @returns
  */
 async function getIncludedItems(request: TestRunRequest, token?: CancellationToken): Promise<TestItem[]> {
-   let testItems: TestItem[] = [];
-   if (request.include) {
-       testItems.push(...request.include);
-   } else {
-       testController?.items.forEach((item: TestItem) => {
-           testItems.push(item);
-       });
-   }
-   if (testItems.length === 0) {
-       return [];
-   }
-   removeTestInvocations(testItems);
-   testItems = await expandTests(testItems, TestLevel.Class, token);
-   // @ts-expect-error
-   const excludingItems: TestItem[] = await expandTests(request.exclude || [], TestLevel.Class, token);
-   testItems = _.differenceBy(testItems, excludingItems, 'id');
-   return testItems;
+    let testItems: TestItem[] = [];
+    if (request.include) {
+        testItems.push(...request.include);
+    } else {
+        testController?.items.forEach((item: TestItem) => {
+            testItems.push(item);
+        });
+    }
+    if (testItems.length === 0) {
+        return [];
+    }
+    if (!handleInvocationRerun(testItems)) { // a single invocation re-run will just update the item
+        removeTestInvocations(testItems);
+    }
+    testItems = await expandTests(testItems, TestLevel.Class, token);
+    // @ts-expect-error
+    const excludingItems: TestItem[] = await expandTests(request.exclude || [], TestLevel.Class, token);
+    testItems = _.differenceBy(testItems, excludingItems, 'id');
+    return testItems;
+}
+
+/**
+ * Check and preparation in case a single invocation of a parameterized test shall be re-run.
+ * @param testItems
+ * @returns true if a single invocation shall be re-run, otherwise false
+ */
+function handleInvocationRerun(testItems: TestItem[]): boolean {
+    let isInvocationRerun: boolean = false;
+
+    // always remove uniqueIds from non-invocation items, since they would have been set for a past run
+    testItems.forEach((item: TestItem) => {
+        const itemData: ITestItemData | undefined = dataCache.get(item);
+        if (itemData && itemData.testLevel !== TestLevel.Invocation) {
+            itemData.uniqueId = undefined;
+        }
+    });
+
+    if (testItems.length === 1 && dataCache.get(testItems[0])?.testLevel === TestLevel.Invocation) {
+        // if a single invocation is to be rerun,
+        // we run the parent method, but with restriction to the single invocation parameter-set
+        const invocation: TestItem = testItems[0];
+        if (!invocation.parent || !dataCache.get(invocation.parent)) {
```

Comment:
Besides `sendError()`, maybe we can also show an error message to let users know there is an error?

```ts
const errMsg = 'Trying to re-run a single test invocation, but could not find a corresponding method-level parent item with data.';
sendError(new Error(errMsg));
window.showErrorMessage(errMsg);
```